editor/src/com/talosvfx/talos/editor/nodes/widgets/CustomVarWidget.java

```diff
@@ -0,0 +1,455 @@
+            color = ColorLibrary.BackgroundColor.MID_GRAY;
+        } else {
+            if (isHover) {
+                color = ColorLibrary.BackgroundColor.BRIGHT_GRAY;
+            }
+        }
+
+        fieldContainer.setBackground(ColorLibrary.obtainBackground(getSkin(), ColorLibrary.SHAPE_SQUARE, color));
+    }
+
+    public void setFieldName(String text) {
+        String oldName = propertyWrapper.propertyName;
+
+        fieldNameLabel.setText(text);
+        fieldNameTextField.setText(text);
+
+        propertyWrapper.propertyName = text;
+
+        fireNameChangedEvent(oldName, text, false);
+    }
+
+    protected void fireValueChangedEvent(boolean isFastChange) {
+        CustomVarWidgetChangeListener.CustomVarChangeEvent nameChangedEvent = Pools.obtain(CustomVarWidgetChangeListener.CustomVarChangeEvent.class);
+        nameChangedEvent.setType(CustomVarWidgetChangeListener.Type.valueChanged);
+        nameChangedEvent.setFastChange(isFastChange);
+
+        try {
+            fire(nameChangedEvent);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            Pools.free(nameChangedEvent);
+        }
+    }
+
+    protected void fireNameChangedEvent(String oldName, String newName, boolean isFastChange) {
+        CustomVarWidgetChangeListener.CustomVarChangeEvent nameChangedEvent = Pools.obtain(CustomVarWidgetChangeListener.CustomVarChangeEvent.class);
+        nameChangedEvent.setType(CustomVarWidgetChangeListener.Type.nameChanged);
+        nameChangedEvent.setOldName(oldName);
+        nameChangedEvent.setNewName(newName);
+        nameChangedEvent.setFastChange(isFastChange);
+
+        try {
+            fire(nameChangedEvent);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            Pools.free(nameChangedEvent);
+        }
+    }
+
+    protected void fireCollapse() {
+        CustomVarWidgetChangeListener.CustomVarChangeEvent nameChangedEvent = Pools.obtain(CustomVarWidgetChangeListener.CustomVarChangeEvent.class);
+        nameChangedEvent.setType(CustomVarWidgetChangeListener.Type.collapse);
+        nameChangedEvent.setFastChange(false);
+
+        try {
+            fire(nameChangedEvent);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            Pools.free(nameChangedEvent);
+        }
+    }
+
+    private void applyValueToWrapper() {
+        innerWidget.applyValueToWrapper(propertyWrapper);
+    }
+
+    @Override
+    public void loadFromXML(XmlReader.Element element) {
+
+    }
+
+    @Override
+    public T getValue() {
+        return propertyWrapper.getValue();
+    }
+
+    @Override
+    public void read(Json json, JsonValue jsonValue) {
+
+    }
+
+    @Override
+    public void write(Json json, String name) {
+
+    }
+
+    public abstract static class CustomVarWidgetChangeListener implements EventListener {
+        public enum Type {
+            nameChanged,
```

Comment:
CAPS for enum values

Reply:
I tried to do similar to libGDX InputListener class 🥺